presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java

```diff
@@ -38,16 +42,18 @@
             implements OperatorFactory
     {
         private final int operatorId;
-        private final ConnectorPageSink pageSink;
+        private final PageSinkManager pageSinkManager;
+        private final WriterTarget target;
         private final List<Integer> inputChannels;
         private final Optional<Integer> sampleWeightChannel;
         private boolean closed;
 
-        public TableWriterOperatorFactory(int operatorId, ConnectorPageSink pageSink, List<Integer> inputChannels, Optional<Integer> sampleWeightChannel)
+        public TableWriterOperatorFactory(int operatorId, PageSinkManager pageSinkManager, WriterTarget writerTarget, List<Integer> inputChannels, Optional<Integer> sampleWeightChannel)
         {
             this.operatorId = operatorId;
             this.inputChannels = checkNotNull(inputChannels, "inputChannels is null");
-            this.pageSink = checkNotNull(pageSink, "pageSink is null");
+            this.pageSinkManager = checkNotNull(pageSinkManager, "pageSinkManager is null");
+            this.target = checkNotNull(writerTarget, "writerTarget is null");
```

Comment:
Verify target is either a `CreateHandle` or `InsertHandle`and throw `IllegalArgumentException` if not.